sympy/core/symbol.py

```diff
@@ -739,38 +742,39 @@ def literal(s):
                 result.append(symbol)
                 continue
 
-            split = _range.split(name)
+            split: List[str] = _range.split(name)
+            split_list: List[List[str]] = []
             # remove 1 layer of bounding parentheses around ranges
             for i in range(len(split) - 1):
                 if i and ':' in split[i] and split[i] != ':' and \
                         split[i - 1].endswith('(') and \
                         split[i + 1].startswith(')'):
                     split[i - 1] = split[i - 1][:-1]
                     split[i + 1] = split[i + 1][1:]
-            for i, s in enumerate(split):
+            for s in split:
                 if ':' in s:
-                    if s[-1].endswith(':'):
+                    if s.endswith(':'):
                         raise ValueError('missing end range')
                     a, b = s.split(':')
                     if b[-1] in string.digits:
-                        a = 0 if not a else int(a)
-                        b = int(b)
-                        split[i] = [str(c) for c in range(a, b)]
+                        a_i = 0 if not a else int(a)
+                        b_i = int(b)
+                        split_list.append([str(c) for c in range(a_i, b_i)])
                     else:
                         a = a or 'a'
-                        split[i] = [string.ascii_letters[c] for c in range(
+                        split_list.append([string.ascii_letters[c] for c in range(
                             string.ascii_letters.index(a),
-                            string.ascii_letters.index(b) + 1)]  # inclusive
-                    if not split[i]:
+                            string.ascii_letters.index(b) + 1)])  # inclusive
+                    if not split_list[-1]:
                         break
                 else:
-                    split[i] = [s]
+                    split_list.append([s])
             else:
                 seq = True
-                if len(split) == 1:
-                    names = split[0]
+                if len(split_list) == 1:
+                    names = split_list[0]
                 else:
-                    names = [''.join(s) for s in product(*split)]
+                    names = [''.join(s) for s in product(*split_list)]
```

Comment:
I do not follow the logic of the rewrites in this part (nor of the original code), but if it passes the tests that should be fine.

One may question if append is better or worse than updating a pre-allocated list, but I guess this is not a performance bottleneck anyway...